contrib/scripts/isc_dhpcd_graphite.py

```diff
@@ -0,0 +1,198 @@
+DEFAULT_CONFIG_FILE = "/etc/dhcpd/dhcpd.conf"
+DEFAULT_CMD_PATH = pathlib.Path("/usr/bin/dhcpd-pools")
+DEFAULT_PROTOCOL = 2  # Python 3: 3, Python 3.8+: 4
+DEFAULT_PORT = "2004"
+
+PICKLE_PROTOCOL = range(0, pickle.HIGHEST_PROTOCOL + 1)
+FLAGS = "-f j"
+METRIC_MAPPER = {
+    "defined": "max",
+    "used": "cur",
+    "touched": "touch",
+    "free": "free",
+}
+
+
+Metric = namedtuple("Metric", ["path", "value", "timestamp"])
+
+
+# parse comand line flags
+def parse_args():
+    parser = argparse.ArgumentParser(description="Send dhcp stats to graphite")
+    parser.add_argument(
+        "server",
+        help="Graphite server to send data to",
+        type=str,
+    )
+    parser.add_argument(
+        "--port",
+        help="Port of Graphite server, if not on 2003 (text) or 2004 (pickle)",
+        type=str,
+    )
+    parser.add_argument(
+        "-f",
+        "--config-file",
+        type=pathlib.Path,
+        help="Complete path to dhcpd-pools config-file. Usually located in /etc/dhcpd/",
+        default=DEFAULT_CONFIG_FILE,
+    )
+    parser.add_argument(
+        "-C",
+        "--command",
+        help="Path to dhcpd-pools command",
+        type=pathlib.Path,
+        default=DEFAULT_CMD_PATH,
+    )
+    parser.add_argument(
+        "-p",
+        "--prefix",
+        help="Path prefix to use for the metric, overriding the default",
+        type=str,
+        default=DEFAULT_PREFIX,
+    )
+    parser.add_argument(
+        "-l",
+        "--location",
+        help="Location, if any, to append to the metric prefix",
+        type=str,
+    )
+    protocol_choices = ("text",) + tuple(str(p) for p in PICKLE_PROTOCOL)
+    parser.add_argument(
+        "-P",
+        "--protocol",
+        help="Protocol to use to send to graphite server",
+        choices=protocol_choices,
+        default=str(DEFAULT_PROTOCOL),
+        type=str,
+    )
+    parser.add_argument(
+        "-n",
+        "--noop",
+        action="store_true",
+        help="Do not send metrics to graphite, print what would be sent instead",
+    )
+    args = parser.parse_args()
+    try:
+        protocol = int(args.protocol)
+    except ValueError:
+        pass
+    else:
+        args.protocol = protocol
+    if not args.port:
+        if isinstance(args.protocol, int):
+            args.port = "2004"
+        else:
+            args.port = "2003"
+    args.actual_prefix = args.prefix
+    if args.location:
+        args.actual_prefix += f".{args.location}"
+    return args
+
+
+# run command and store json output
+def exec_dhcpd_pools(config_file, cmd_path=DEFAULT_CMD_PATH):
+    flags = f"-c {config_file} {FLAGS}".split()
+    cmd = [cmd_path] + list(flags)
+    result = subprocess.run(cmd, capture_output=True)
+    if result.returncode:
+        sys.exit(result.stderr)
+    return json.loads(result.stdout)
+
+
+# reformat the data
+def _clean_vlan(location):
+    regex = re.search("vlan\d+", location)
+    return regex.match
```

Comment:
This script crashes with an exception on this line when I try to run it:

```pytb
Traceback (most recent call last):
  File "isc_dhpcd_graphite.py", line 198, in <module>
    main()
  File "isc_dhpcd_graphite.py", line 187, in main
    output = render(jsonblob, args.actual_prefix, args.protocol)
  File "isc_dhpcd_graphite.py", line 144, in render
    return _render_pickle(jsonblob, prefix, protocol)
  File "isc_dhpcd_graphite.py", line 159, in _render_pickle
    input = _tuplify(jsonblob, prefix)
  File "isc_dhpcd_graphite.py", line 134, in _tuplify
    vlan = _clean_vlan(vlan_stat["location"])
  File "isc_dhpcd_graphite.py", line 126, in _clean_vlan
    return regex.match
AttributeError: 're.Match' object has no attribute 'match'
```

Reply:
Fixed.